src/eigen/lobpcg_hyper_impl.jl

```diff
@@ -227,64 +227,71 @@ function LOBPCG(A, X, B=I, precon=((Y, X, R)->R), tol=1e-10, maxiter=100; ortho_
 
             # Form Rayleigh-Ritz subspace
             if niter > 2
-                Y = hcat(X,R,P)
-                AY = hcat(AX,AR,AP)
-                BY = hcat(BX,BR,BP)
+                Y = hcat(X, R, P)
+                AY = hcat(AX, AR, AP)
+                if B != I
+                    BY = hcat(BX, BR, BP)
+                else
+                    BY = Y
+                end
             else
-                Y = hcat(X,R)
-                AY = hcat(AX,AR)
-                BY = hcat(BX,BR)
+                Y = hcat(X, R)
+                AY = hcat(AX, AR)
+                if B != I
+                    BY = hcat(BX, BR)
+                else
+                    BY = Y
+                end
             end
-
-
             cX, λs = RR(Y, AY, BY, M-nlocked)
 
             # Update X. By contrast to some other implementations, we
             # wait on updating P because we have to know which vectors
             # to lock (and therefore the residuals) before computing P
             # only for the unlocked vectors. This results in better
             # convergence.
-            new_X = Y*cX
-            # cX is orthogonal, so there is no accuracy loss there
-            new_AX = AY*cX
+            new_X  = Y * cX
+            new_AX = AY* cX  # cX is orthogonal, so no accuracy loss here
             if B != I
-                new_BX = BY*cX
+                new_BX = BY * cX
             else
                 new_BX = new_X
             end
         end
 
         ### Compute new residuals
-        R = new_AX - new_BX*Diagonal(λs)
+        new_R = new_AX - new_BX * Diagonal(λs)
```

Comment:
why the rename?

Reply:
no not mix it up with the old `R` (was needed for the lazy concatenation that I tried) also follows the analogy with P and X.